index.d.ts

```diff
@@ -0,0 +1,344 @@
+  save: (params?: GetParams) => Promise<FeathersVuexModel<ModelType>>;
+  patch: (params?: GetParams) => Promise<FeathersVuexModel<ModelType>>;
+  update: (params?: GetParams) => Promise<FeathersVuexModel<ModelType>>;
+  clone: () => Promise<FeathersVuexModel<ModelType>>;
+  reset: () => Promise<FeathersVuexModel<ModelType>>;
+  remove: (params?: GetParams) => Promise<FeathersVuexModel<ModelType>>;
+}
+
+export type FeathersVuexModel<T> = FeathersVuexModelClass<T> & T;
+
+interface FeathersVuexModelOptions {
+  isClone?: boolean;
+  skipCommit?: boolean;
+}
+
+/**
+ * Vue Plugin
+ */
+type FeathersVuexGlobalModelsIndex<T> = {
+  [P in keyof T]: {
+    new(data?: Partial<T[P]>, options?: FeathersVuexModelOptions): FeathersVuexModel<T[P]>;
+    find: (params?: FindParams) => FeathersVuexModel<T[P]>[];
+    findInStore: (params?: FindParams) => FeathersVuexModel<T[P]>[];
+    get: (id: any, params?: GetParams) => FeathersVuexModel<T[P]>[];
+    getFromStore: (id: any, params?: GetParams) => FeathersVuexModel<T[P]>[];
+  };
+}
+
+/**
+ * $FeathersVuex: FeathersVuexGlobalModelsIndex<Services> = {users: WrapFeathersVuexModel()}
+ * ->
+ * $FeathersVuex = {
+ *   users: typeof FeathersVuexModel<T['users']>
+ * }
+ *
+ * new $FeathersVuex.users()
+ */
+
+// This doesn't work because merging this into a string signature isn't a valid type
+// Leaving it here so that hopefully someone can figure it out in the future
+// type FeathersVuexGlobalModelsByPath<T> = {
+//   byServicePath: FeathersVuexGlobalModelsIndex<T>
+// }
+
+type FeathersVuexGlobalModels<T = any, P = any> = FeathersVuexGlobalModelsIndex<T> // & FeathersVuexGlobalModelsByPath<P>; // See comment above
+
+/**
+ * @description The type for the $FeathersVuex vue plugin. To type this stronger include the
+ * following in a typings.d.ts file
+ *
+ *
+ * @example
+ * declare module "vue/types/vue" {
+ *   import { FeathersVuexGlobalModels } from "feathers-vuex";
+ *   interface Vue {
+ *     $FeathersVuex: FeathersVuexGlobalModels<Services>;
+ *   }
+ * }
+ *
+ * @description Where services is something like this
+ * @example
+ * interface Services {
+ *   users: User
+ *   //...
+ * }
+ *
+ * interface User {
+ *   name: string;
+ *   rating: number;
+ *   //...
+ * }
+ * @description
+ * The second generic is for the "byServicePath" property, and can be used in a similar way, just make a
+ * ServicesByPath interface, where the key for the service is the path, not the name
+ *
+ * @example
+ * interface Services {
+ *   usersPath: User
+ *   //...
+ * }
+ *
+ * declare module "vue/types/vue" {
+ *   import { FeathersVuexGlobalModels } from "feathers-vuex";
+ *   interface Vue {
+ *     $FeathersVuex: FeathersVuexGlobalModels<Services, ServicesByPath>;
+ *   }
+ * }
+ *
+ */
+interface FeathersVuexPluginType extends FeathersVuexGlobalModels { }
+
+declare module "vue/types/vue" {
+  interface Vue {
+    // $FeathersVuex: FeathersVuexPluginType;
```

Comment:
Same here.

Reply:
This should be an commented. The line before is needed.

Reply:
Fixed